### DISCUSSION TITLE
What's the proper way to gracefully and safely shut down an entire k3s cluster in preparation for a physical move?

### DISCUSSION THREAD

**Discussion by matthewadams** [k3s-io/k3s#9865] [Q&A]
Hi,

We've seen https://docs.k3s.io/upgrades/killall#k3s-service but aren't clear that this is the safest, most graceful way to shutdown the whole cluster.  We are prepping to physically move the three physical nodes to another location.  By safe & graceful, I mean that we won't lose any data on volumes, we won't corrupt dbs, etc.

Also, for good measure, what would be the proper way to bring the cluster back up?

Like, we'd love a `k3s` server and/or agent command like `k3s shutdown` or similar that gracefully brings down the whole kit & caboodle, preserving all data, etc.  This seems like a problem unique to on-prem clusters, so it seems warranted for `k3s` to provide a convenient means to achieve shutdown & restart.

**Comment by brandond** [comment]
Just drain the nodes in whatever order your application prefers to handle terminating the workload, then run the killall script to clean up the rest.
